sdk/docs/core/README.md

```diff
@@ -143,10 +143,10 @@ Note however that cancellation is performed as a best effort; it is not guarante
 
    ```C
    // Some function creates a child with a 10-second expiration:
-   az_context child = az_context_with_expiration(&az_context_app, 10);
+   az_context child = az_context_create_with_expiration(&az_context_app, 10);
```

Comment:
Looking at this again, this might warrant an update to the guidelines, because it recommends the format:
`az_<shortname>_<objname>_create_<noun>`
https://azure.github.io/azure-sdk/clang_design.html#clang-apisurface-standardized-verbs

That format works when the `noun` we are creating is different from the obj creating it, like `az_iot_create_client`
or `az_keyvault_keyclient_create_key`

In our case here, for us to follow the guidelines, we would need something like:
`az_context_create_context_with_expiration`

And that might be OK too, though seeing `context` twice in the method name is a bit cumbersome.

cc @barcharcraz

Reply:
It also means `az_span_create_span_from_str`. I'd add a rule so that we can omit object name when it is a constructor, i.e. the object it creates is the same. But I'd go as we decide. If we don't reach a decision here within a day, I'd remove this rename from this PR.

Reply:
I think we shouldn't go that route, because should we apply that naming rule, `az_span_create` would be named `az_span_create_span`, and it wont be possible to name a function `az_X_create` - it will always have to be `az_X_create_X`. And later overloads will be `az_X_create_X_with_Y` / `_from_Y`.

Reply:
@barcharcraz , are you fine with `az_X_create`/`az_X_create_with_Y` instead of `az_X_create_X`/`az_X_create_X_with_Y`?
If we don't reach a conclusion soon, I'll leave the naming as it is, so it conforms to neither naming scheme :)

Reply:
I think that was the original intent of the wording actually. I'll respond with more detail after reviewing the rest of the PR.

Reply:
Thank you! You don't have to review the entire PR, but certainly that would be super awesome! :)

Reply:
ah, the reason for the confusion is because these "create" function are constructors not member functions. There are separate guidelines for constructors that do what you suggest, there should be a cross-referencing link, however.

Reply:
So, to clarify more specifically, the name `az_context_create_with_expiration()` is ok. Correct?

Reply:
Can you share a link? I couldn't find the separate guideline on constructors.

Reply:
Given the answer is "yes", I crated a work item to update guidelines: https://github.com/Azure/azure-sdk-for-c/issues/999

Reply:
I just pinged Charlie via Teams, the answer is yes. I am resolving this comment thread. We have a work item to track guidelines update.

Reply:
> I crated a work item to update guidelines: #999

One more to a thousand! 4 digit issues and PRs :)